packages/react-components/src/templates/TagsPageBody.tsx

```diff
@@ -56,11 +66,25 @@ const TagsPageBody: React.FC<TagsPageBodyProps> = ({
       />
     }
   >
-    {results.map((data) => (
-      <div key={data.id}>
-        <SharedResearchCard {...data} />
-      </div>
-    ))}
+    {results.map((data) => {
+      if ('documentType' in data) {
+        return (
+          <div key={data.id}>
+            <SharedResearchCard {...data} />
+          </div>
+        );
+      }
+
+      if ('expertiseAndResourceTags' in data) {
```

Comment:
Should this be a type discriminator?

Reply:
I think this is related to https://github.com/yldio/asap-hub/pull/3856#discussion_r1375980474 right? if so it was not done properly for similar reasons. But I'm happy to accept suggestion of how to do it properly if there's a nice way to make TS work :)